### PR TITLE
[Fixes 7139] Add format filter mapping for xml formatters

### PR DESCRIPTION
**Pull request by kichalla**

### CODE REVIEW

File: test/Microsoft.AspNetCore.Mvc.Formatters.Xml.Test/Internal/MvcXmlDataContractSerializerMvcOptionsSetupTest.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Logging.Abstractions;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Formatters.Xml.Internal
+{
+    public class MvcXmlDataContractSerializerMvcOptionsSetupTest
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.Formatters.Xml.Test/Internal/MvcXmlDataContractSerializerMvcOptionsSetupTest.cs]: Could you add a functional test for this?

File: src/Microsoft.AspNetCore.Mvc.Formatters.Xml/Internal/MvcXmlDataContractSerializerMvcOptionsSetup.cs [hunk 2]
```diff
@@ -42,6 +43,16 @@ public void Configure(MvcOptions options)
             options.OutputFormatters.Add(new XmlDataContractSerializerOutputFormatter(_loggerFactory));
             options.InputFormatters.Add(new XmlDataContractSerializerInputFormatter(options));
 
+            // Do not override any user mapping
+            var key = "xml";
+            var mapping = options.FormatterMappings.GetMediaTypeMappingForFormat(key);
```
Comment (mkArtakMSFT) [inline src/Microsoft.AspNetCore.Mvc.Formatters.Xml/Internal/MvcXmlDataContractSerializerMvcOptionsSetup.cs]: Is the "FormatterMappings" property guaranteed to have value?
If not, please rewrite as:
options.FormatterMappings?.GetMediaTypeMappinForFormat(key)

File: src/Microsoft.AspNetCore.Mvc.Formatters.Xml/Internal/MvcXmlDataContractSerializerMvcOptionsSetup.cs [hunk 3]
```diff
@@ -42,6 +43,16 @@ public void Configure(MvcOptions options)
...
+            if (string.IsNullOrEmpty(mapping))
+            {
+                options.FormatterMappings.SetMediaTypeMappingForFormat(
+                    key,
+                    MediaTypeHeaderValue.Parse("application/xml"));
```
Comment (mkArtakMSFT) [inline src/Microsoft.AspNetCore.Mvc.Formatters.Xml/Internal/MvcXmlDataContractSerializerMvcOptionsSetup.cs]: Regarding these literals, do we always just hardcode these? Isn't there a set of constants for those?

Comment (kichalla) [reply src/Microsoft.AspNetCore.Mvc.Formatters.Xml/Internal/MvcXmlDataContractSerializerMvcOptionsSetup.cs]: Good point. I found that we already have something similar that I could use.

File: src/Microsoft.AspNetCore.Mvc.Formatters.Xml/Internal/MvcXmlDataContractSerializerMvcOptionsSetup.cs (hunk 2 above)
Comment (kichalla) [reply src/Microsoft.AspNetCore.Mvc.Formatters.Xml/Internal/MvcXmlDataContractSerializerMvcOptionsSetup.cs]: We new up this property when MvcOptions is [new'd up](https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/MvcOptions.cs#L38), so unless until a user explicitly sets this to null, it shouldn't be a problem. If a user explicitly sets to null, they are shooting themselves in the foot? (they could clear existing formats if they do not want format mappings)

**Pull request merged**